src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -163,6 +165,10 @@ Status NrfWiFiDriver::AddOrUpdateNetwork(ByteSpan ssid, ByteSpan credentials, Mu
     VerifyOrReturnError(ssid.size() <= sizeof(mStagingNetwork.ssid), Status::kOutOfRange);
     VerifyOrReturnError(credentials.size() <= sizeof(mStagingNetwork.pass), Status::kOutOfRange);
 
+    // It is forbidden to change network credentials during associating process.
```

Comment:
Should it rather be checked in `ConnectNetwork`? This function only updates the local credentials object. What shouldn't happen is calling WPA supplicant API while processing the previous request, right?

Reply:
Yes... I guess this came from wrong understanding of what this method should really do. Fixed.